analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -287,6 +295,27 @@ class SpdxDocumentFile(
             }
         }
 
+    /**
+     * Return [PackageLinkage] of [dependency] if there is relationship in [relationships] that matches the [dependency]
+     * and relationship's [partner]. By default [PackageLinkage.DYNAMIC] is returned if no relationship is found.
+     */
+    private fun getLinkageFromDependency(
+        relationships: List<SpdxRelationship>,
+        dependency: SpdxPackage,
+        partner: String?
+    ): PackageLinkage {
+        val spdxId = dependency.spdxId
+
+        val linkageRelationship =
+            relationships.singleOrNull {
+                it.relatedSpdxElement == spdxId &&
+                        SPDX_LINKAGE_RELATIONSHIPS.keys.contains(it.relationshipType) &&
+                        it.spdxElementId == partner
+            }
+
+        return SPDX_LINKAGE_RELATIONSHIPS[linkageRelationship?.relationshipType] ?: PackageLinkage.DYNAMIC
+    }
```

Comment:
How about this instead:

    ) =
        relationships.mapNotNull { relation ->
            SPDX_LINKAGE_RELATIONSHIPS[relation.relationshipType]?.takeIf {
                relation.spdxElementId == partner && relation.relatedSpdxElement == dependency.spdxId
            }
        }.singleOrNull() ?: PackageLinkage.DYNAMIC